backend/src/controllers/filter-item-controller.ts

```diff
@@ -0,0 +1,166 @@
+            }
+
+            tempData.forEach((data) => {
+                const isIdValid = mongoose.Types.ObjectId.isValid(data._id) 
+    
+                if (!isIdValid) {
+                    return res.status(500).json({ message: "Invalid Data" }) 
+                }
+                
+                if (data.name.toLowerCase().includes(name.toLowerCase()) && data.itemOwner.toLowerCase().includes(itemOwner.toLowerCase())) {
+                    tempDataObject = data._id
+                }  
+            })
+            fetchedInventoryItems = {itemId: tempDataObject, size: size, amount: {$gt: 1}}
+        } else if (itemOwner && !size && !name) {
+            let tempData: any[] = await Item.find({})
+            let tempDataObject: any;
+            if (!tempData) {
+                res.status(404).json({ message: "Not Found" })
+            }
+
+            tempData.forEach((data) => { 
+                const isIdValid = mongoose.Types.ObjectId.isValid(data._id) 
+    
+                if (!isIdValid) {
+                    return res.status(500).json({ message: "Invalid Data" }) 
+                }
+
+                if (data.itemOwner.toLowerCase().includes(itemOwner.toLowerCase())) {
+                    tempDataObject = data._id
+                }  
+            })
+            fetchedInventoryItems = {itemId: tempDataObject, amount: {$gt: 1}}
+        } else if (!itemOwner && size && !name) {
+            fetchedInventoryItems = await InventoryItem.find({size: size})
+        } else if (!itemOwner && !size && name) {
+            let tempData: any[] = await Item.find({})
+            let tempDataObject: any;
+            if (!tempData) {
+                res.status(404).json({ message: "Not Found" })
+            }
+
+            tempData.forEach((data) => {
+                const isIdValid = mongoose.Types.ObjectId.isValid(data._id) 
+    
+                if (!isIdValid) {
+                    return res.status(500).json({ message: "Invalid Data" }) 
+                }
+                
+                if (data.name.toLowerCase().includes(name.toLowerCase())) {
+                    tempDataObject = data._id
+                }  
+            })
+            fetchedInventoryItems = {itemId: tempDataObject, amount: {$gt: 1}}
+        } else if (itemOwner && name && !size) {
+            let tempData: any[] = await Item.find({})
+            let tempDataObject: any;
+            if (!tempData) {
+                res.status(404).json({ message: "Not Found" })
+            }
+
+            tempData.forEach((data) => {
+                const isIdValid = mongoose.Types.ObjectId.isValid(data._id) 
+    
+                if (!isIdValid) {
+                    return res.status(500).json({ message: "Invalid Data" }) 
+                }
+                
+                if (data.name.toLowerCase().includes(name.toLowerCase()) && data.itemOwner.toLowerCase().includes(itemOwner.toLowerCase())) {
```

Comment:
Compare `itemOwner` with an `===` operator because it is an ID, so `itemOwner`'s ID must perfectly match the ID in `data`